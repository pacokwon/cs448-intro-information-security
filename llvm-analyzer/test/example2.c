#include "homework.h"

int main() {
  int x = source();
  int z;
  if (x > -2) {
    z = 10 / x; // error
    if (x < -2) {
      z = 10 / x; // safe
    }
  } else {
    z = 10 / x; // safe
  }
  z = 10 / x; // error
  return 0;
}
