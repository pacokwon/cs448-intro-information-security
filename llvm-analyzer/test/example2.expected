Potential Division-by-zero @ example2.c:main:7:12, %call = Top
Potential Division-by-zero @ example2.c:main:14:10, %call = Top
